roles/rsyslog/roles/input_roles/files/defaults/main.yml

```diff
@@ -64,45 +46,52 @@ __rsyslog_conf_imfile_inputs:
 
       - comment: 'Log messages from log files'
         options: |-
-          # FIXME: Need to normalize a log before passing it to elasticsearch output
-          {% if rsyslog_elasticsearch is defined and rsyslog_elasticsearch %}
-
-          {% for element in rsyslog_logs_collections %}
-          {% if element.name is defined and element.type == "files" %}
-          {% if element.rsyslog_input_log_path is defined %}
-          {% if element.rsyslog_input_log_tag is defined %}
-          input(type="imfile" file="{{ element.rsyslog_input_log_path }}" tag="{{ element.rsyslog_input_log_tag }}" ruleset="try_es")
-          {% else %}
-          input(type="imfile" file="{{ element.rsyslog_input_log_path }}" tag="{{ rsyslog_input_log_tag }}" ruleset="try_es")
-          {% endif %}
-          {% else %}
-          {% if element.rsyslog_input_log_tag is defined %}
-          input(type="imfile" file="{{ rsyslog_input_log_path }}" tag="{{ element.rsyslog_input_log_tag }}" ruleset="try_es")
-          {% else %}
-          input(type="imfile" file="{{ rsyslog_input_log_path }}" tag="{{ rsyslog_input_log_tag }}" ruleset="try_es")
-          {% endif %}
-          {% endif %}
-          {% endif %}
+          {% for input in logging_inputs %}
+          {%   if input.type == "files" %}
+          input(type="imfile" file="{{ input.rsyslog_input_log_path | d(rsyslog_input_log_path) }}" tag="{{ input.name }}")
+          {%   endif %}
           {% endfor %}
-
-          {% else %}
-
-          {% for element in rsyslog_logs_collections %}
-          {% if element.name is defined and element.type == "files" %}
-          {% if element.rsyslog_input_log_path is defined %}
-          {% if element.rsyslog_input_log_tag is defined %}
-          input(type="imfile" file="{{ element.rsyslog_input_log_path }}" tag="{{ element.rsyslog_input_log_tag }}")
-          {% else %}
-          input(type="imfile" file="{{ element.rsyslog_input_log_path }}" tag="{{ rsyslog_input_log_tag }}")
-          {% endif %}
-          {% else %}
-          {% if element.rsyslog_input_log_tag is defined %}
-          input(type="imfile" file="{{ rsyslog_input_log_path }}" tag="{{ element.rsyslog_input_log_tag }}")
-          {% else %}
-          input(type="imfile" file="{{ rsyslog_input_log_path }}" tag="{{ rsyslog_input_log_tag }}")
-          {% endif %}
-          {% endif %}
-          {% endif %}
+          {% set ns = namespace(files_output = false) %}
+          {% set rsyslog_outputs = logging_outputs | d([]) %}
+          {% for output in rsyslog_outputs %}
+          {%   if output.type == "files" %}
+          {%     set ns.files_output = true %}
+          {%   endif %}
           {% endfor %}
-
+          {% if not ns.files_output %}
+          {%   set rsyslog_outputs = rsyslog_outputs + [{"name": "files", "type": "files"}] %}
           {% endif %}
```

Comment:
You don't need the temporary variable here.  You can search the list directly like this:
```
          {% if not rsyslog_outputs|selectattr("type","==","files")|list %}
          {%   set rsyslog_outputs = rsyslog_outputs +[{"name": "files", "type": "files"}] %}
          {% endif %}
```
`selectattr` is horribly documented :-( - found through trial and error
also found through trial and error - you cannot modify `logging_outputs` - the copy `{% set rsyslog_outputs = logging_outputs | d([]) %}` is necessary in order to have something to append to

Reply:
Thanks, @richm!  Verified your proposal.